plugin-process-to-cgroup-bridge/src/transform.rs

```diff
@@ -0,0 +1,199 @@
+//  Context Production: Will return a PathBuf with value "/proc"
+//
+//  Context Tests:      Will return a PathBuf with the value inside the static PROC_PATH variable
+//                      This value can be injected by tests using `set_proc_path_for_test` function
+//                      Note that this variable can be set only once
+#[cfg(test)]
+use std::sync::OnceLock;
+
+#[cfg(test)]
+static PROC_PATH: OnceLock<PathBuf> = OnceLock::new();
+
+#[cfg(test)]
+fn get_proc_path() -> &'static PathBuf {
+    PROC_PATH.get_or_init(|| {
+        panic!("PROC_PATH not initialized in test - call set_proc_path_for_test() first");
+    })
+}
+
+#[cfg(test)]
+pub fn set_proc_path_for_test(path: PathBuf) {
+    let _ = PROC_PATH.set(path);
+}
+
+#[cfg(not(test))]
+fn get_proc_path() -> PathBuf {
+    PathBuf::from("/proc")
+}
+// --- END OF TESTS PURPOSE ---
+
+pub struct ProcessToCgroupBridgeTransform {
+    processes_metrics_ids: Vec<RawMetricId>,
+    merge_similar_cgroups: bool,
+    keep_processed_measurements: bool,
+}
+
+impl Transform for ProcessToCgroupBridgeTransform {
+    fn apply(&mut self, measurements: &mut MeasurementBuffer, _ctx: &TransformContext) -> Result<(), TransformError> {
+        let mut cgroup_measurements = MeasurementBuffer::new();
+        let mut kept_measurements = MeasurementBuffer::new();
+        for measurement in measurements.iter() {
+            if let Some(cgroup_measurement) = self.cgroup_measurement_from_measurement(measurement)? {
+                cgroup_measurements.push(cgroup_measurement);
+                if self.keep_processed_measurements {
+                    kept_measurements.push(measurement.clone());
+                }
+            } else {
+                kept_measurements.push(measurement.clone());
+            }
+        }
+        measurements.clear();
+
+        if self.merge_similar_cgroups {
+            let mut aggregated_cgroup_measurements = Self::aggregate_cgroups_measurements(&mut cgroup_measurements);
+            cgroup_measurements.clear();
+            cgroup_measurements.merge(&mut aggregated_cgroup_measurements);
+        }
+
+        measurements.merge(&mut kept_measurements);
+        measurements.merge(&mut cgroup_measurements);
+
+        Ok(())
+    }
+}
+
+impl ProcessToCgroupBridgeTransform {
+    pub fn new(
+        processes_metrics_ids: Vec<RawMetricId>,
+        merge_similar_cgroups: bool,
+        keep_processed_measurements: bool,
+    ) -> Self {
+        Self {
+            processes_metrics_ids,
+            merge_similar_cgroups,
+            keep_processed_measurements,
+        }
+    }
+
+    fn cgroup_measurement_from_measurement(
+        &self,
+        measurement: &MeasurementPoint,
+    ) -> anyhow::Result<Option<MeasurementPoint>> {
+        if !self.processes_metrics_ids.contains(&measurement.metric) {
+            return Ok(None);
+        }
+        let pid = match Self::extract_process_id_from_measurement(measurement) {
```

Comment:
If you ignore the error of `extract_process_id_from_measurement`, it should probably return `Option` instead.